### PR TITLE
Use a local server in tests

### PR DESCRIPTION
**Pull request by little-bobby-tables**
Hello! I'd like to thank you for this project, it seems to suit my use cases and avoids some mistakes I probably would've made had I had to do it myself (namely, downloading the whole file in memory instead of streaming it).

If you don't mind, I suggest you to use a local server for the test suite. This allows you to test corner cases by fine tuning the response size, status, and headers, not to mention it doesn't require an Internet connection in development.

I was also slightly confused by `"Uncomment the line below to check the real size"` in code. I think it can be safely removed, because apparently HTTPoison won't exceed the size reported by Content-Length anyway (see the test case I've added).

### CODE REVIEW

**Comment by asiniy** [comment]
@little-bobby-tables thanks for PR! I'm a little bit busy right now, please wait for approx. a couple of days. Will response you soon

**Comment by little-bobby-tables** [comment]
Okay, sure.

File: lib/download.ex [hunk 1]
```diff
@@ -116,8 +116,6 @@ defmodule Download do
 
   defp handle_async_response_chunk(%AsyncEnd{}, opts), do: finish_download({ :ok }, opts)
 
-  # Uncomment one line below if you are prefer to test not "Content-Length" header response, but a real file size
```
Comment (asiniy) [inline lib/download.ex:119]: The only problem that I found.

Please write some tests for this case - when `Content-Length` header and real body size aren't match.
Ok, we both know that right now we have `httpoison` behind `download` interface.

But what will be if we replace `httpoison` with, saying, `httpc`? Do you know how will it behave this case?

This case have to be tested too.

/cc @little-bobby-tables

Comment (bravely) [reply lib/download.ex:119]: Hey there, looking to get this package updated as I can definitely use a powerful but easy to use download library!

I'm not @little-bobby-tables, but it looks like the test [here](https://github.com/asiniy/download/pull/2/files#diff-a61b801e8dbe5c11dd374c223f983765R51) tests when the Content-Length header and real body size don't match- it downloads only the Content-Length, without anything else. Is another behavior desirable?

I'm not understanding the second point:

> Ok, we both know that right now we have httpoison behind download interface.

> But what will be if we replace httpoison with, saying, httpc? Do you know how will it behave this case?

> This case have to be tested too.

Nothing in his test code really relies on the library being HTTPoison, at least in this PR. What were you looking for here?
